Add source docs, update GIFs, add template repo docs (#197)
kubeshop/botkube-docs: docs/plugin/local-testing.md

**Prerequisite**

- [Node.js](https://nodejs.org/en/download/), to use `npx`

**Steps**

1. Follow steps from [`CONTRIBUTING.md`](https://github.com/kubeshop/botkube/blob/main/CONTRIBUTING.md#build-and-run-locally) about running Botkube locally.

2. Create a file with your plugins' repository, plugin configuration and bindings for enabled communication platform:

   ```yaml
   plugins:
     repositories:
       local-repo:
         url: http://localhost:8080/plugins-index.yaml

   executors:
     "plugin-based":
       local-repo/executor-name: # Plugin name syntax: <repo>/<plugin>[@<version>]. If version is not provided, the latest version from repository is used.
         enabled: true
         config: {} # Plugin's specific configuration.
   sources:
     "plugin-based":
       local-repo/source-name: # Plugin name syntax: <repo>/<plugin>[@<version>]. If version is not provided, the latest version from repository is used.
         enabled: true
         config: {} # Plugin's specific configuration.

   communications:
     # Enable a given communication platform and define bindings to a given executor and source plugins.
   ```

>    For example, for Slack and example `echo` and `ticker` plugins provide `appToken` and `botToken` and create `/tmp/config-values.yaml` with the following values:

Reviewer: If this is actually a script, then maybe this is better:
```suggestion
   For example, for Slack and example `echo` and `ticker` plugins, provide `appToken` and `botToken` and run the script:
```

   <details>
     <summary>Create /tmp/config-values.yaml</summary>

   ```yaml
   cat << EOF > /tmp/config-values.yaml
   plugins:
     repositories:
       local-repo:
         url: http://localhost:8080/plugins-index.yaml

   executors:
     "plugin-based":
       local-repo/echo:
         enabled: true
         config:

>            changeResponseToUpperCase: true

Reviewer: ```suggestion
           transformResponseToUpperCase: true
```

   sources:
     "plugin-based":
       local-repo/ticker:
         enabled: true
         config:

>            duration: 5s

Reviewer: ```suggestion
           interval: 5s
```

   plugins:
     repositories:
       local-repo:
         url: http://localhost:8080/plugins-index.yaml

   executors:
     "plugin-based":
       local-repo/executor-name: # Plugin name syntax: <repo>/<plugin>[@<version>]. If version is not provided, the latest version from repository is used.
         enabled: true
         config: {} # Plugin's specific configuration.
   sources:
     "plugin-based":
       local-repo/source-name: # Plugin name syntax: <repo>/<plugin>[@<version>]. If version is not provided, the latest version from repository is used.
         enabled: true
         config: {} # Plugin's specific configuration.

   communications:
     # Enable a given communication platform and define bindings to a given executor and source plugins.
   ```

   For example, for Slack and example `echo` and `ticker` plugins, run:
   :::note
   Do not forget to fill the `appToken` and `botToken` properties.
   :::

   <details>

>      <summary>Create /tmp/config-values.yaml</summary>

Author: Actually, I removed the `Edit the comm_config.yaml file.`. As later, I suggest to use:
```bash
export BOTKUBE_CONFIG_PATHS="${BOTKUBE_REPO_PATH}/helm/botkube/values.yaml,/tmp/config-values.yaml"
```

   ```yaml
   cat << EOF > /tmp/config-values.yaml
   plugins:
     repositories:
       local-repo:
         url: http://localhost:8080/plugins-index.yaml

   executors:
     "plugin-based":
       local-repo/echo:
         enabled: true
         config:
           changeResponseToUpperCase: true
   sources:
     "plugin-based":
       local-repo/ticker:
         enabled: true
         config:
           duration: 5s
   communications:
     default-group:
       socketSlack:
         enabled: true
         channels:
           default:
             name: random
             bindings:
               executors:
                 - 'plugin-based'
               sources:
                 - 'plugin-based'
         appToken: "" # provide your token starting with xapp-1-

>          botToken: "" # provide your token starting with xoxb-

Reviewer: Please also add:

```yaml
settings:
  clusterName: local-dev
```
To make the output messages look a bit better 😄

   EOF
   ```

   </details>

3. Start static plugins' server:

>    ```bash

Author: the `go run test/helpers/plugin_server.go` is a bit different and it builds and returns the index on each query, so the only thing is to only rebuild the binaries. As a result, it's a bit faster. Later IMO we should just create some CLI under `/hack` or `/cmd` that will help with all those plugin testing things and use it in all places 👍

   ```

8. Build plugins and start Botkube:

   :::note
   Each time you change a plugin source code, re-run the above command.
   :::

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<div className="tab-container-nested">
<Tabs>
  <TabItem value="raw" label="For scratch projects" default>

1. Download index builder:

```bash
go get github.com/kubeshop/botkube/hack
```

2. Build plugins and run Botkube:

```bash
# rebuild plugins only for current GOOS and GOARCH
goreleaser build --rm-dist --snapshot --single-target &&
# regenerate index
go run github.com/kubeshop/botkube/hack -binaries-path "./dist" -url-base-path "http://localhost:8080/dist" &&
# remove cached plugins
rm -rf $BOTKUBE_PLUGINS_CACHE__DIR &&
# start Botkube
${BOTKUBE_REPO_PATH}/botkube
```

  </TabItem>
  <TabItem value="repo" label="For projects created from template repository">

```bash
# rebuild plugins only for current GOOS and GOARCH
make build-plugins-single &&
# regenerate index
env PLUGIN_DOWNLOAD_URL_BASE_PATH="http://localhost:8080/dist" make gen-plugin-index &&
# remove cached plugins
rm -rf $BOTKUBE_PLUGINS_CACHE__DIR &&
# start Botkube

> ${BOTKUBE_REPO_PATH}/botkube

Reviewer: There wasn't any Botkube binary building step - I'd suggest changing it to
```suggestion
cd ${BOTKUBE_REPO_PATH} && go run cmd/botkube/main.go
```
